src/plugins/main.ts

```diff
@@ -172,9 +167,9 @@ export const Enhance = (
  * Allows plugins to update their state after a move / event.
  */
 export const Flush = (state: State, opts: PluginOpts): State => {
-  // Note that we flush plugins in reverse order, to make sure that plugins
-  // that come before in the chain are still available.
-  [...DEFAULT_PLUGINS, ...opts.game.plugins].reverse().forEach((plugin) => {
+  // We flush the events plugin first, then custom plugins and the core plugins
+  // This means custom plugins cannot use the events API but will be available in event hooks
+  [PluginEvents, ...opts.game.plugins, ...CORE_PLUGINS].forEach((plugin) => {
```

Comment:
Currently, it is possible to build custom plugins on top of each other, something like:

```js
const PluginA = {
  name: 'A',
  api: () => ({
    getFive: () => 5,
  }),
};

const PluginB = {
  name: 'B',
  setup: ({ ctx }) => ({
    n: ctx.A.getFive(),
  }),
  flush: ({ ctx, data }) => ({
    n: data.n + ctx.A.getFive(),
  }),
};

const game = {
  plugins: [PluginA, PluginB],
};
```

Here `PluginA` provides some API methods, which `PluginB` consumes. Without reversing the user-provided plugins, this breaks during `flush`, because `PluginA` will be removed before `PluginB.flush` is called.

To continue support for this pattern we should probably keep `reverse` here, like this:

```suggestion
  [...CORE_PLUGINS, ...opts.game.plugins, PluginEvents].reverse().forEach((plugin) => {
```

Reply:
Makes sense!